devdocs/dependencies.md

```diff
@@ -27,3 +27,12 @@ To propagate the update of dependencies to your Eclipse configuration, run the f
 `./gradlew resetEclipseDeps`
 
 Sometimes, the changes from this command might not show up in Eclipse immediately. "Refreshing" the project or restarting Eclipse should fix that.
+
+# Managing Client-Side Dependencies
+
+Currently, the dependencies for CSS/JS are hosted in the repository and listed down in `src/main/webapp/package.json`.
+
+When adding new libraries, try to find the library in the [npm registry](https://www.npmjs.com) and if it can be found, list it in the aforementioned package file, then download and include the necessary library files (usually in `/dist` or `/build` folder) in the repository.
+Node.js developers can do this with the command `npm install --save package@version`. Remember to exclude any non-exact version syntax (e.g `^` and `~`) in the package file as we require specific versions.
+
+Otherwise, simply host the necessary library files in the repository without updating the package file.
```

Comment:
2nd option meaning using dependencies not listed in NPM? It's not a possibility to dismiss, but I think the chances are slim at best. Also a library not listed in NPM may not be that reliable (does not mean the converse is true). Should I make looking for files in NPM registry a must?

As for the benefit, this addition has no benefit on its own. When we integrate VersionEye (going to be done immediately after having this PR merged), it can benefit from getting info on outdated libraries, license violations, and security issues ([here's a sample](https://www.versioneye.com/user/projects/578f4d2c5181bd004a1ff6de) on how this package file can be used by VersionEye). More benefits can be explored if we decide to use more of NPM/Node.js features, but this would require further research/discussions.